modules/cms/controllers/Index.php

```diff
@@ -59,6 +61,31 @@ public function __construct()
     {
         parent::__construct();
 
+        Event::listen('backend.form.extendFieldsBefore', function ($widget) {
+            if (!$widget->getController() instanceof Index) {
+                return;
+            }
+            if (!$widget->model instanceof CmsCompoundObject) {
+                return;
+            }
+            if (key_exists('code', $widget->secondaryTabs['fields']) && CmsHelpers::safeModeEnabled()) {
+                // save code field config
+                $codeField = $widget->secondaryTabs['fields']['code'];
+                unset($widget->secondaryTabs['fields']['code']);
+
+                // add hint field first
+                $widget->secondaryTabs['fields']['hint'] = [
+                    'tab' => 'cms::lang.editor.code',
+                    'type' => 'hint',
+                    'path' => '~/modules/cms/controllers/index/_hint_field.htm',
+                    'cssClass' => 'p-b-0',
+                ];
```

Comment:
@mjauvin the other thing that you could do is include this field in the default form, but mark it as hidden: true and then just unhide it under the proper conditions.

Reply:
I thought about it but then I'd have to include it in 4 seperate forms... which do you prefer?

Reply:
I'm fine with it being in the separate fields.yaml